### PR TITLE
synchronizer updates

### LINKED ISSUE #117: Refactor synchronizer to use on chain trees

### PR DESCRIPTION
**Pull request by vimwitch**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`yarn build`) was run locally and any changes were pushed
- [x] Lint (`yarn lint --check`) has passed locally and any fixes were made for failures

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Closes #117

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

### CODE REVIEW

**Comment by vimwitch** [comment]
This is ready for review. Right now I'm generating epoch tree inclusion proofs on chain. This means we can't build #105 because we need to keep all the old epoch trees around for making proofs. I think it's probably worth emitting events so the synchronizer can build the epoch tree offchain to avoid this, but it should be a new task.

File: packages/core/src/Synchronizer.ts [hunk 1]
```diff
@@ -1045,225 +520,39 @@ export class Synchronizer extends EventEmitter {
...
-        const tree = await this.genEpochTree(epoch)
         console.log(`Epoch ${epoch} ended`)
         db.upsert('Epoch', {
             where: {
                 number: epoch,
             },
             update: {
                 sealed: true,
-                epochRoot: tree.root.toString(),
+                // epochRoot: tree.root.toString(),
```
Comment (vivianjeng) [inline packages/core/src/Synchronizer.ts:543]: it can be removed

File: packages/core/src/Synchronizer.ts [hunk 2]
```diff
@@ -1045,225 +520,39 @@ export class Synchronizer extends EventEmitter {
...
             },
             create: {
                 number: epoch,
                 sealed: true,
-                epochRoot: tree.root.toString(),
+                // epochRoot: tree.root.toString(),
```
Comment (vivianjeng) [inline packages/core/src/Synchronizer.ts:548]: same here

**Review by vivianjeng — APPROVED** [review]
LGTM

**Pull request merged**